src/plugins/UI-plugin.js

```diff
@@ -0,0 +1,26 @@
+import { EventEmitter } from "events";
+
+export default class UIPlugin extends EventEmitter {
```

Comment:
It may be worth making these changes to the template-plugin and inheriting from it in the case of waterfall-plugin. And you can rename template-plugin to ui-plugin.)

Reply:
ok, got it